client/components/chat-list.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import NewChatForm from './new-chat-form';
+
+export default class ChatList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formIsOpen: false,
+      form: {
+        chatName: '',
+        userName: ''
+      }
+    };
+    this.openNewChatForm = this.openNewChatForm.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.closeForm = this.closeForm.bind(this);
+    this.submitForm = this.submitForm.bind(this);
+    this.resetForm = this.resetForm.bind(this);
+  }
+
+  openNewChatForm() {
+    const openForm = Object.assign({}, this.state);
+    openForm.formIsOpen = true;
+    this.setState(openForm);
+  }
+
+  handleChange(target) {
+    const newState = Object.assign({}, this.state);
+    if (target.id === 'chat-name') {
+      newState.form.chatName = target.value;
+      this.setState(newState);
+      return;
+    }
+    newState.form.userName = target.value;
```

Comment:
Close. It's more complicated than this though. You are modifying an object within an object, so you need to copy that too.
```js
const newState = Object.assign({}, this.state, {
  form: Object.assign({}, this.state.form, {
    chatName: target.value
  }
})
```